### PR TITLE
PP-1149: Adjust the sleep time for TestSoftWalltime.test_soft_held

### PR DESCRIPTION
**Pull request by vstumpf**
#### Bug/feature Description
* TestSoftWalltime.test_soft_held is failing.
* [PP-1149](https://pbspro.atlassian.net/browse/PP-1149)

#### Affected Platform(s) and PBS Version
* All Platforms

#### Cause / Analysis / Design
* TestSoftWalltime.test_soft_held is failing due to a timing issue.

#### Solution Description
* Increase the sleep time during the test to ensure the estimated softwalltime is correct.

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__
[1149-SmokeTest-20180126091647.txt](https://github.com/PBSPro/pbspro/files/1668592/1149-SmokeTest-20180126091647.txt)
[1149-softwalltime-20180126085324.txt](https://github.com/PBSPro/pbspro/files/1668593/1149-softwalltime-20180126085324.txt)

### CODE REVIEW

**Comment by bhroam** [comment]
The changes you made seem fine.  I understand that this isn't your test, but I question the validity of it.  It submits a soft walltime job and waits for it to extend.  The scheduler sets estimated.soft_walltime to 14.  The job is held and rerun.  The estimated.soft_walltime doesn't change.  The hold is released.  The estimated.soft_walltime is still 14 because it doesn't get reset when the job is requeued.  We sleep another 10s to wait for the estimated.soft_walltime to extend again.  The problem is the test is checking for 21.  The soft_walltime should start again.  Sleeping 10s should extend it again to 14.  The only way this test is passing is if some weird race condition is allowing the job to run another 5s before the new cycle.

**Comment by vstumpf** [comment]
So the last sleep should be for 15 seconds, right?
The current test passes because there's no repeat limit in the expect, so it waits the extra 5 seconds in trying to match.

**Comment by bhroam** [comment]
I guess you have two options.  You can specifically unset estimated.soft_walltime or you can sleep for 15 (or maybe 17) seconds.  The longer sleep is probably more straightforward.

**Comment by anamikau** [comment]
Thanks Vincent. Looking at the test again I agree with Bhroam. Like you mentioned increase the sleep time and check for estimated.soft_walltime

**Comment by vstumpf** [comment]
I'll increase the sleep to 17 and reduce the max_attempts to 5 for the final expect.

File: test/tests/functional/pbs_soft_walltime.py [hunk 1]
```diff
@@ -880,9 +880,12 @@ def test_soft_held(self):
 
         # Wait for some more time and verify that soft_walltime
         # extending again
-        time.sleep(7)
+        self.logger.info(
+            "Sleep to let soft_walltime get extended again")
```
Comment (anamikau) [inline test/tests/functional/pbs_soft_walltime.py]: since we have learnt that why 17s is needed could you make the log message bit more explicit

**Comment by vstumpf** [comment]
I got 2 sign-offs, rebased.

It seems like appveyor can't build python for whatever reason, I'll wait for Monday until asking for a rebuild.

**Comment by mike0042** [comment]
Appveyor build failed.

**Comment by vstumpf** [comment]
I've rebased, this should be ready to be merged.

**Pull request merged**